docs/guides/cloudeventssource.md

```diff
@@ -0,0 +1,120 @@
+# Receiving External CloudEvents
+
+The Trigermesh `CloudEventsSource` API object is used to ingest Cloudevents produced from external sources.
+
+## Configuring a CloudEventsSource Object
+
+The CloudEventsSource accepts parameters to set authentication, URL path and rate limiter. When succesfuly created it exposes an endpoint to listen for CloudEvents.
+
+### Configuring Credentials (Optional)
+
+Credentials can be configured using [Basic Authentication](https://datatracker.ietf.org/doc/html/rfc2617) using Kubernetes secrets to manage passwords.
+
+Credentials are defined as arrays, allowing clients to use multiple user/password items.
+
+The credentials are defined under `spec.credentials.basicAuths`:
+
+```yaml
+spec:
+  credentials:
+    basicAuths:
+    - username: user1
+      password:
+        valueFromSecret:
+          name: password1
+          key: password
+    - username: user2
+      password:
+        valueFromSecret:
+          name: password2
+          key: password
+```
+
+### Configuring Path (Optional)
+
+The `spec.path` parameter is used configure the URL path where CloudEvents will be accepted.
+
+```yaml
+spec:
+  path: /mypath
+```
+
+### Configuring Rate Limiter (Optional)
+
+Rate Limiter is used to filter the quantity of requests per second that an adapter instance can receive. When the configured limit per time window is reached, HTTP code 429 is returned along information on when the client should retry.
+
+```console
+HTTP/1.1 429 Too Many Requests
+retry-after: 1650204469582011930
+```
+
+To configure the Rate Limiter use the `spec.rateLimiter.requestsPerSecond` parameter:
+
+```yaml
+spec:
+  rateLimiter:
+    requestsPerSecond: 1000
+```
+
+It must be noted when configuring the Rate Limiter that:
+
+- The CloudEventsSource is based on Knative Services and under load will create many adapters. The rate limiter value is set individualy per adapter instance.
+- A low value of the rate limiter might prevent the adapter from scaling.
+
+### Configuring CloudEvents Sink
+
+The `spec.sink` parameter is a [Destination](https://pkg.go.dev/knative.dev/pkg/apis/duck/v1#Destination) that points to an object or URL that will receive the ingested CloudEvents.
```

Comment:
the https://pkg.go.dev/knative.dev/pkg/apis/duck/v1#Destination link is very weird, this just looks like a bug to me. shall we remove it?

Reply:
Maybe we need a general ticket to get rid of this everywhere because i think it is show on every component in the reference docs.

Reply:
I think reference should have it, but here ... it depends. For people unaware of Knative it should not so I am removing it.

There are some Knative technicalities around when we go down the specs for structures we use from upstream, I think we can hide them for regular non-knative users, but they still need to be there to explain how our components can be built.